src/main/java/org/broadinstitute/hellbender/engine/ReadsDataSource.java

```diff
@@ -134,7 +138,49 @@ public ReadsDataSource( final List<Path> samPaths, final List<Path> samIndices )
      * @param customSamReaderFactory SamReaderFactory to use, if null a default factory with no reference and validation
      *                               stringency SILENT is used.
      */
-    public ReadsDataSource( final List<Path> samPaths, final List<Path> samIndices, SamReaderFactory customSamReaderFactory ) {
+    public ReadsDataSource( final List<Path> samPaths, final List<Path> samIndices,
+        SamReaderFactory customSamReaderFactory) {
+        this(samPaths, samIndices, customSamReaderFactory, 0, 0);
+    }
+
+    /**
+     * Initialize this data source with multiple SAM/BAM/CRAM files, explicit indices for those files,
+     * and a custom SamReaderFactory.
+     *
+     * @param samPaths paths to SAM/BAM/CRAM files, not null
+     * @param samIndices indices for all of the SAM/BAM/CRAM files, in the same order as samPaths. May be null,
+     *                   in which case index paths are inferred automatically.
+     * @param customSamReaderFactory SamReaderFactory to use, if null a default factory with no reference and validation
+     *                               stringency SILENT is used.
+     * @param cloudPrefetchBuffer MB size of caching/prefetching wrapper for the data, if on Google Cloud (0 to disable).
+     * @param cloudIndexPrefetchBuffer MB size of caching/prefetching wrapper for the index, if on Google Cloud (0 to disable).
+     */
+    public ReadsDataSource( final List<Path> samPaths, final List<Path> samIndices,
+            SamReaderFactory customSamReaderFactory,
+            int cloudPrefetchBuffer, int cloudIndexPrefetchBuffer) {
+        this(samPaths, samIndices, customSamReaderFactory,
+            (cloudPrefetchBuffer > 0 ? is -> ReadsDataSource.addPrefetcher(cloudPrefetchBuffer, is)
+                                     : Function.identity()),
+            (cloudIndexPrefetchBuffer > 0 ? is -> ReadsDataSource.addPrefetcher(cloudIndexPrefetchBuffer, is)
```

Comment:
`ReadsDataSource.addPrefetcher` -> `addPrefetcher`, since we're now inside `ReadsDataSource`